### ISSUE TITLE
swiper-avy usage

### ISSUE THREAD

**Issue by joaocortes** [abo-abo/swiper#1915]
I'm running emacs 26.1 with `counsel`, `ivy`, `avy`and `swiper` installed from melpa and up to date.
My init file is (excluding `use-package` configurations)
``` elisp
(use-package avy)     
(avy-setup-default)
     (global-set-key (kbd "C-c C-j") 'avy-resume)
     (global-set-key (kbd "C-'") 'avy-goto-char-2)

(use-package counsel
  :init (ivy-mode 1)
  (setq ivy-use-virtual-buffers t)
  (setq ivy-count-format "(%d/%d) "))

(global-set-key (kbd "C-s") 'swiper)
```
When searching for "a" with `swiper` through this buffer,
``` text
a x a x a
a e a e a
```
, I expected a call to `swiper-avy` would allow me to jump to any of the matches. 
Instead, `swiper-avy` only sets up one shortcut for each matching line.
Is it something wrong with my config?

Thanks

**Comment by avocadoras** [comment]
I have a similar issue but when I try to invoke avy in the minibuffer it tells me to evaluate a Lisp Expression
```
(use-package avy
    :after ivy
    :ensure t
    :bind (("M-s" . avy-goto-word-1)
	   ("M-g g" . avy-goto-line)))
(use-package ivy
  :ensure t
  :bind
  (:map ivy-mode-map ("C-'" . ivy-avy))
  :config
  (ivy-mode 1)
  (setq ivy-use-virtual-buffers t)
  (setq ivy-count-format "%d/%d ")
  (setq ivy-display-style 'fancy))

(use-package ivy-rich
  :ensure t
  :after ivy
  :init
  (setq ivy-rich-path-style 'abbrev
	ivy-virtual-abbreviate 'full)
  :config (ivy-rich-mode 1))

(use-package counsel
  :ensure t
  :after ivy
  :config
  (counsel-mode))

(use-package swiper
  :ensure t
  :after ivy
  :bind (("C-s" . swiper)
	 ("C-c C-r" . ivy-resume)))

;; Ivy Actions
;;;;;;;;;;;;;;;

;;;  delete files when perusing your file system
;; (ivy-set-actions
;;  'counsel-find-file
;;  '(("d" delete-file "delete")))
```
`ivy-resume` also does not work. When I press the `C-c C-r` I  get nothing...

**Comment by abo-abo** [comment]
Thanks, please test.

**Comment by avocadoras** [comment]
@abo-abo the fix works now but it's a bit spotty. I cannot get it to work in a python file that only has these lines
```py
 d = "helo"




a = "hi"

sfr  = " hlloww"
```
I cannot also get it to work in an Org buffer. 

I don't have any extra packages for Python (not using elpy only company/company-jedi). Checked the bindings and nothing binded to `C-'` in company, so swiper-avy should work.

Also, nothing happens when I press `C-c C-r` in an org buffer, but sometimes works in an Emacs-Lisp buffer.

In summary, I have tested in an Emacs-Lisp file and it worked once. I tested in a Python and Org mode buffer and still have the same issue. I made sure I updated the package by
1. `M-x list-packages`
2. `C-s swiper`
3. `Shift-u`

And also went into `swiper.el` to line 242 and evaluated the function.

**Comment by abo-abo** [comment]
@avocadoras You don't need any extra config. Just have `avy` installed and press <kbd>C-'</kbd>.

The problem is likely in your config. Try to reproduce with `emacs -Q`, load only `swiper` and `avy`.

**Comment by joaocortes** [comment]
In my case It's solved.
Thank you

**Comment by avocadoras** [comment]
I loaded `emacs -Q` then I copied this into scratch and evaluated each. expression. Still cannot get `swiper-avy` to work. Also, `ivy-avy` has some issues and opens another buffer.

```elisp
(require 'package)
(setq package-enable-at-startup nil)

(add-to-list 'package-archives
	     '("melpa" . "https://melpa.org/packages/"))

(package-initialize)

;; Bootstrap 'use-package'
(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))

(use-package avy
  :ensure t
  :bind ("M-s" . avy-goto-word-1)) ;; also accepts integer input like go-to-line

(use-package counsel
  :ensure t
  :config
  (counsel-mode))

(use-package ivy
  :ensure t
  :after (counsel avy)
  :bind ("C-c C-r" . ivy-resume)
  :config
  (ivy-mode 1)
  (setq ivy-use-virtual-buffers t)
  (setq ivy-count-format "%d/%d ")
  (setq ivy-display-style 'fancy))

(use-package swiper
  :ensure t
  :after (ivy avy)
  :bind ("C-s" . swiper))

  (use-package ivy-rich
    :ensure t
    :after ivy
    :init
    (setq ivy-rich-path-style 'abbrev
	  ivy-virtual-abbreviate 'full)
    :config (ivy-rich-mode 1))
```
Although the image here was not loaded with `emacs -Q`, I observe the same behavior when I repeat the steps stated above. 
![ivy-avy-no-work-fin](https://user-images.githubusercontent.com/36335687/52360480-c6ca6e80-2a01-11e9-8857-5b4167ccef11.gif)

**Comment by abo-abo** [comment]
@avocadoras I'm glad you solved your problem :)
